app/scripts/lib/app-start.js

```diff
@@ -609,7 +610,21 @@ define(function (require, exports, module) {
         error, self._sentryMetrics, self._metrics, self._window);
     },
 
+    /**
+     * A requireOnDemand that loads modules on demand.
+     *
+     * @param {string} module - name of module to load.
+     */
+    requireOnDemand (module) {
```

Comment:
This was done to facilitate unit testing. A more obvious way would be to pass in a mock `requireOnDemand` on initialization, if that's not passed in, use the real version.